AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOperation.swift

```diff
@@ -152,23 +165,25 @@ final class InitialSyncOperation: AsynchronousOperation {
             return
         case .success(let queryResult):
             syncQueryResult = queryResult
+            startedAt = syncQueryResult.startedAt
         }
 
-        let items = syncQueryResult.items
+        items.append(contentsOf: syncQueryResult.items)
```

Comment:
Append every items into the array first until not more nextToken and finally offer the items to `AWSModelReconciliationQueue` at once. This way, we can make sure the `recordsReceived` is final and no race condition between `recordsReceived` and `reconciledReceived`. 
And according to (https://github.com/aws-amplify/amplify-ios/pull/758#discussion_r486463032), I don't think count is a brittle way so far. I can't think of a better way to track the completion of `ModelReconciliation`. Perhaps we can find another time to find a way

Reply:
This approach means we're not processing *any* reconciliations until we get *all* of the reconciliations. That's not good practice for an asynchronous process, and I especially don't want to negatively impact our sync performance just so we can emit Hub events.